pkg/controller/atlasproject/auditing.go

```diff
@@ -3,96 +3,87 @@ package atlasproject
 import (
 	"reflect"
 
-	"go.mongodb.org/atlas/mongodbatlas"
-
-	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/audit"
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
 	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
 )
 
-func ensureAuditing(workflowCtx *workflow.Context, project *akov2.AtlasProject) workflow.Result {
-	result := createOrDeleteAuditing(workflowCtx, project.ID(), project)
-	if !result.IsOk() {
-		workflowCtx.SetConditionFromResult(api.AuditingReadyType, result)
-		return result
-	}
-
-	if isAuditingEmpty(project.Spec.Auditing) {
-		workflowCtx.UnsetCondition(api.AuditingReadyType)
-		return workflow.OK()
-	}
-
-	workflowCtx.SetConditionTrue(api.AuditingReadyType)
-	return workflow.OK()
+type auditController struct {
```

Comment:
Great!

Reply:
indeed, this is a slick, small, and neat little state machine done here 😍